### DISCUSSION TITLE
Mark all as read and max items

### DISCUSSION THREAD

**Discussion by damonachey** [gandf/slick-rss#54] [General]
When I have max items set to say 100 (default), if the feed has 150 actual items, and I do mark all as read it only marks the 100 as read and leaves the 50 oldest unread (and not viewable).  Is there any way of removing that 50 also without having to show all the items?

I'd like to add, this is more for performance.  I've noticed things are really slow when there are lots of feeds.  I have 92 feeds in my list that have around 1300 posts total.

**Comment by gandf** [comment]
Hi,

### Max Items

With max item, parser stop to parse after max item reached. This means you don't have to waste time with huge feeds.
Hashes of other feeds are not calculated and cannot be marked as read.
I don't know what to do about this because if I read the whole feed, I'll slow down those with large feeds.

### Performance

Since move to Manifest V3, severals issue persists.
As there was no longer access to an HTML interpreter, I had to use a library that transforms HTML into json and then make an engine that parses this json. This takes a lot more time.

Full information is only available after parsing all the feeds. As a result, viewer page does not display everything correctly immediately.

To prevent the viewer from overwriting information in the DB, I've made sure that most viewer requests go through the service worker.
But because it's monothreaded, the service worker isn't always available to execute the viewer request.
What's more, Chrome sometimes puts the worker service to sleep.
As a result, viewer requests are not always executed to save information in DB.

I think that to make app more faster, I must rewrite structured data in DB to allow direct write from viewer.
There's another point that may explain the slowness.
Redirected or error feeds take much longer than others.
I put red warnings at the top of the page when this happens. It's better to change the feed address to get the right one.
Some feeds have very high server response times.

**If you don't have personal link in your feeds, you can send it export from your feeds list so I can look at which feed is slowing everything down and determine why.**
